src/openApi/openApiWrapper.ts

```diff
@@ -66,7 +77,7 @@ export default class OpenApiWrapper {
             return new ApiResponse(error.statusCode, {
               title: error.message,
               details: error.details,
-              errorId: requestLogger.invocationId,
+              errorId: requestLogger.invocationId, // TODO: Keep?
```

Comment:
Can the `correlationId` be set to headers in this case as well?

Reply:
Oh yes, missed this one. Also if you have an opinion - the headers can be passed to `ApiResponse`'s constructor too. Or we could add `correlationId: string` parameter to the constructor to not leak the header name. I used the `withCorrelationId` everywhere as there is that one case where we don't construct the ApiResponse so we need a way to append the correlation ID

Reply:
It will be annoying to always provide this when creating ApiResponse.. when we really only want to append the `correlationId` in the middlewares, right?

I think it is okay.